### PR TITLE
[feature] Add progenitor information to meta

### PR DESCRIPTION
**Pull request by syou6162**
## Background
If you use dbt-osmosis to maintain column descriptions for a long period of time, you often experience the following.

> I want to modify the description of this column to a better description. However, the description of this column seems to have been propagated by dbt-osmosis. Which model column description should I modify?

It would be helpful if this information could be kept statically in yaml instead of at runtime. For this reason, the `--add-progenitor-to-meta` option has been added in this pull request (Default value is `False` so as not to change the existing behavior).

## Example
For example, suppose you have the following data source

```yaml
version: 2
sources:
  - name: my_dataset
    database: my-project
    tables:
      - name: order
        columns:
          - name: my_description
            description: "This is description"
```

`dbt-osmosis yaml refactor --add-progenitor-to-meta ... ` will output the following yaml. You can include information about which model the column description is derived from in `meta.osmosis_progenitor`. This information is very useful because it tells you which model to modify when modifying column descriptions.

```yaml
version: 2
models:
  - name: mart_order
    columns:
      - name: my_description
        description: "This is description"
        meta:
          osmosis_progenitor: source.my_project.my_dataset.order
```

This example is very simple, but when models are linked in multiple layering using dbt, it is difficult to quickly determine which model description is the progenitor. The `meta.osmosis_progenitor` is very useful because it tells you which model to modify when modifying a column description.

### CODE REVIEW

File: src/dbt_osmosis/main.py [hunk 1]
```diff
@@ -241,6 +259,9 @@ def organize(
     check: bool = False,
     models: Optional[List[str]] = None,
     skip_add_columns: bool = False,
+    skip_add_tags: bool = False,
```
Comment (syou6162) [inline src/dbt_osmosis/main.py:262]: This line is not directly related to this pull request, but I forgot to add it when I was working on https://github.com/z3z1ma/dbt-osmosis/pull/78, so I'm fixing it here together 🙏 .

File: src/dbt_osmosis/core/osmosis.py [hunk 2]
```diff
@@ -1005,6 +1007,9 @@ def update_undocumented_columns_with_prior_knowledge(
                 node.columns[column].replace(kwargs={"name": column, **prior_knowledge})
             for model_column in yaml_file_model_section["columns"]:
                 if model_column["name"] == column:
+                    if self.add_progenitor_to_meta:
+                        prior_knowledge.setdefault("meta", {})
+                        prior_knowledge["meta"]["osmosis_progenitor"] = progenitor
```
Comment (syou6162) [inline src/dbt_osmosis/core/osmosis.py:1012]: `osmosis` is given in the prefix so that it can be seen as related to osmosis.

**Comment by syou6162** [comment]
@z3z1ma Could you review this pull request?

**Comment by z3z1ma** [comment]
LGTM
